nautobot/core/tests/test_commands.py

```diff
@@ -80,3 +95,22 @@ def test_virtualchassis_failure(self):
 
         with self.assertRaises(CommandError):
             self.run_command()
+
+    def test_permission_constraints_failure(self):
+        """Test permission constraints logs warning when needed for CustomField."""
+        device_ct = ContentType.objects.get_for_model(Device)
+        site_ct = ContentType.objects.get_for_model(Device)
+        custom_field = CustomField.objects.create(type=CustomFieldTypeChoices.TYPE_TEXT, name="custom_field")
+        custom_field.content_types.add(site_ct)
```

Comment:
I am a little confused here. Do we need line 102 and line 104 here?
We are trying to add `site_ct` which is actually the ContentType for Device?
The warning should be raised just by including `site` in the constraints of `obj_perm` right?

Reply:
Good catches. Definitely 102 should be `get_for_model(Site)` if anything. But it'd be interesting to remove lines 102 through 104 and see if that affects the test at all, especially given that the custom_field has name `custom_field` but the obj_perm is using `custom_field_in` (probably meant to be `custom_field__in` instead?)...